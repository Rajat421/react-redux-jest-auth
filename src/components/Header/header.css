header ul{
    padding: 0;
    margin: 10px;
    text-align: right;
}
header ul li{
    list-style: none;
    display: inline-block;
    position: relative;
}
header ul li a {
    text-decoration: none;
    color: #000;
    font-size: 14px;
    padding: 10px;
    display: block;
    letter-spacing: 1px;
}
header ul li ul{
    display: none;
    position: absolute;
    top: 25px;
    right: -10px;
    z-index: 5;
    text-align: left;
    margin-right: 0;
}
header ul li ul li{
    display: block;
    cursor: pointer;
    background-color: #fff;
}
header ul li:hover ul{
    display: block;
}
header ul li i{
    cursor: pointer;
    padding: 10px;
    font-size: larger;
}
