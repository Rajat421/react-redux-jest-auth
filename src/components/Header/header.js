
import React,{Component} from 'react'
import {Link} from 'react-router-dom'
import './header.css'

class Header extends Component{
    render(){
        return(
            <header className="header">
                <ul>
                    <li> <Link to="/home">Home</Link></li>
                    <li> <Link to="/aboutus">About</Link>
                        <ul>
                            <li><Link to="/aboutus/profile">Profile</Link></li>
                            <li><Link to="/aboutus/team">Team</Link></li>
                            <li><Link to="/aboutus/contactus">Contact</Link></li>
                        </ul>
                    </li>
                    <li><i className="fa fa-user fa-lg" aria-hidden="true"></i>
                        <ul>
                            <li onClick={this.props.logout}>Logout</li>
                        </ul>
                    </li>
                </ul>
            </header>

        )
    }
}

export default Header

