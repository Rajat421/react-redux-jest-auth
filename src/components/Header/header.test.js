
import React from "react";
import ReactDOM from 'react-dom'
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Header from "./header";
import {MemoryRouter} from "react-router";

test('Header component should render as expected',()=>{
    const component =shallow(<MemoryRouter><Header/></MemoryRouter>);
    const tree =toJson(component);
    expect(tree).toMatchSnapshot()
});

test('Header should load without crahing',()=>{
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter ><Header/></MemoryRouter>, div);
});

