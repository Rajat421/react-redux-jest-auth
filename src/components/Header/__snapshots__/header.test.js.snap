// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "90p7lv",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "90p7lv",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Header />
</Router>
`;
